src/test/java/uk/gov/ida/eventemitter/EventEmitterTest.java

```diff
@@ -1,68 +1,112 @@
 package uk.gov.ida.eventemitter;
 
 import org.joda.time.DateTime;
-import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
 
 import java.io.ByteArrayOutputStream;
+import java.io.IOException;
 import java.io.PrintStream;
 import java.util.UUID;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
 
+@RunWith(MockitoJUnitRunner.class)
 public class EventEmitterTest {
 
-    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    private static final String ENCRYPTED_EVENT = "encrypted event";
+    private static final UUID ID = UUID.randomUUID();
+    private static final String TIMESTAMP = "2018-02-06T14:37:55.467Z";
+    private static final String EVENT_TYPE = "Error Event";
 
     private EventEmitter eventEmitter;
     private EventEmitter eventEmitterWithoutRecordingSystem;
+    private TestEvent event;
+
+    @Mock
+    private Encrypter encrypter;
+
+    @Mock
+    private SqsClient sqsClient;
 
     @Before
-    public void setUp() {
-        System.setOut(new PrintStream(outContent));
+    public void setUp() throws Exception {
+        when(encrypter.encrypt(any(Event.class))).thenReturn(ENCRYPTED_EVENT);
 
-        eventEmitter = new EventEmitter(true);
-        eventEmitterWithoutRecordingSystem = new EventEmitter(false);
-    }
+        eventEmitter = new EventEmitter(true, encrypter, sqsClient);
+        eventEmitterWithoutRecordingSystem = new EventEmitter(false, encrypter, sqsClient);
 
-    @After
-    public void tearDown() {
-        System.setOut(System.out);
+        event = new TestEvent(ID, DateTime.parse(TIMESTAMP), EVENT_TYPE);
     }
 
     @Test
-    public void recordMethodShouldWriteEventDetailsToSystemOut() {
-        final UUID id = UUID.randomUUID();
-        final String timestamp = "2018-02-06T14:37:55.467Z";
-        final String eventType = "Error Event";
-        final TestEvent event = new TestEvent(id, DateTime.parse(timestamp), eventType);
-
-        eventEmitter.record(event);
-
-        assertThat(outContent.toString())
-            .isEqualTo(String.format("Event ID: %s, Timestamp: %s, Event Type: %s\n", id, timestamp, eventType));
+    public void shouldLogEventDetailsWhenEventEmitterIsConfiguredToSendToSqs() throws IOException {
```

Comment:
These tests never check if anything is sent to SQS. (they just check the logging)
Should we really be checking the 'send to SQS' instead / as well?